lib/routes/content.js

```diff
@@ -86,41 +72,18 @@ const routes = application => {
      * @param {object} next - Express callback
     */
     app.get(`/${config.content.base}/:type/${config.content.actions.add}`, (req, res, next) => {
-      const type = utils.singleItem('id', req.params.type.toLowerCase(), types);
-      const errors = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].errors`, {});
-      const values = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].content`, {});
-      const workflow = workflows.workflow(type, allflows, config, req);
-
-      _.unset(req.session, 'form.content.add');
+      const good = utils.content.check.url(req, types, allFlows);
 
-      // check type exists
-      if (!utils.content.check.type(req, type)) {
-        return next();
-      }
+      if (typeof good === 'string') {
+        _.set(req.session, '404', {
+          message: good,
+          safe: `/${config.content.base}`,
+        });
 
-      // check workflow exists
-      if (!workflow) {
         return next();
       }
 
-      const steps = (JSON.parse(JSON.stringify(workflow.steps))).reverse();
-      const step = steps[steps.length - 1];
-
-      return utils.fill(types, type, references, database).then(ct => {
-        return content.only(req.params.type.toLowerCase(), values, [ct]).then(merged => {
-          return content.form(merged, errors).then(form => {
-            res.render('content/add', {
-              form,
-              action: req.url.replace(config.content.actions.add, config.content.actions.save),
-              type: ct,
-              config: config.content,
-              step,
-            });
-          });
-        }).catch(e => {
-          next(e);
-        });
-      });
+      return routes.add(req, res, good.type, good.workflow, references, types);
```

Comment:
As we control the signature for this function, why not just have it take `found` instead of `type` and `workflow` and have `routes.add` pick out `type` and `workflow` for itself?